src/sentry/tasks/integrations/github/language_parsers.py

```diff
@@ -216,6 +216,118 @@ def generate_function_name_conditions(function_names: list[str], stack_frame: in
         )
 
 
+class PHPParser(LanguageParser):
+    issue_row_template = "| **`{function_name}`** | [**{title}**]({url}) {subtitle} <br> `Event Count:` **{event_count}** |"
+
+    @staticmethod
+    def extract_functions_from_patch(patch: str) -> set[str]:
+        r"""
+        Type of function declaration    Example
+        Function declaration:           public static function hello(argument1, argument2)
+        Anonymous Function:             $hello = function(argument1, argument2)
+        Arrow function:                 $arrowFunc = fn($parameter) => $parameter + 1;
+        """
+
+        function_declaration_regex = r"^@@.*@@[^=]*?\s*function\s+(?P<fnc>[^\(]*)\(.*$"
+        anonymous_function_regex = r"^@@.*@@.*\s+\$(?P<fnc>\w+)\s*=\s*function\s*\(.*$"
+        arrow_function_regex = r"^@@.*@@.*\s+\$(?P<fnc>\w+)\s*=\s*fn\([^)]*\)\s*=>.*$"
+
+        regexes = [
+            function_declaration_regex,
+            arrow_function_regex,
+            anonymous_function_regex,
+        ]
+
+        functions = set()
+        for regex in regexes:
+            functions.update(set(re.findall(regex, patch, flags=re.M)))
+
+        return functions
+
+    @staticmethod
+    def _get_function_name_conditions(stackframe_level: int, function_names: list[str]):
+        """
+        For PHP we need a special case of matching both for the function name itself and for
+        "::" + the function name, because sometimes Snuba stores the function name as "namespace::FunctionName".
+        """
+        prepended_function_names = ["%::" + function_name for function_name in function_names]
+        function_name_conditions = [
```

Comment:
Is there an opportunity here to abstract these into the parent class instead of copy pasting?

Reply:
hmm i agree, there is some shared logic here that can be abstracted. maybe we have a list of possible prefixes as a class variable and then we can move these underscore functions into the base class